src/ToDo.js

```diff
@@ -1,21 +1,29 @@
 import React from "react";
 
 const ToDo = ({ todo, handleToggle }) => {
+  const [checked, setChecked] = React.useState(todo.complete);
   const handleClick = (e) => {
+    setChecked(e.currentTarget.checked);
     e.preventDefault();
     handleToggle(e.currentTarget.id);
```

Comment:
in fact, in this line you can do `todo.id` instead of getting the id from the element. Then you don't have to assign the id to both the checkbox and the div tag.